chemclipse/plugins/org.eclipse.chemclipse.chromatogram.xxd.peak.detector.supplier.firstderivative/src/org/eclipse/chemclipse/chromatogram/xxd/peak/detector/supplier/firstderivative/core/PeakDetectorMSD.java

```diff
@@ -179,10 +181,8 @@ public IProcessingInfo detect(IChromatogramSelectionMSD chromatogramSelection, I
 					peak.setDetectorDescription(DETECTOR_DESCRIPTION);
 					peaks.add(peak);
 				}
-			} catch(IllegalArgumentException e) {
-				logger.warn(e);
-			} catch(PeakException e) {
-				logger.warn(e);
+			} catch(Exception e) {
+				logger.warn(e.getLocalizedMessage(),e);
```

Comment:
Instead of logging it would be better to add a Patameter of type MessageConsumer and add an errorMessage to the consumer then so the error could be propagated to the UI.
It would also be a good idea instead of passing the message as is to add a more descriptive message (e.g. 'creating peak for at retention time X failed because of <exception message>')